src/storage/timeLog.ts

```diff
@@ -0,0 +1,34 @@
+export class TimeLog {
+  constructor(
+    public startTime: string,
+    public duration: number,
+    public endTime: string,
+    public fileName: string,
+    public codeEditor: string,
+    public osName: string,
+    public projectName: string,
+    public programmingLanguage: string
+  ) {}
+
+  static create(data: {
+    startTime: string;
+    duration: number;
+    endTime: string;
+    fileName: string;
+    codeEditor: string;
+    osName: string;
+    projectName: string;
+    programmingLanguage: string;
+  }): TimeLog {
+    return new TimeLog(
+      data.startTime,
+      data.duration,
+      data.endTime,
+      data.fileName,
+      data.codeEditor,
+      data.osName,
+      data.projectName,
+      data.programmingLanguage
+    );
+  }
+}
```

Comment:
Let's try this implementation: 
```suggestion
export interface TimeLogData {
  startTime: string;
  duration: number;
  endTime: string;
  fileName: string;
  codeEditor: string;
  osName: string;
  projectName: string;
  programmingLanguage: string;
}

export class TimeLog {
  private constructor(
    public readonly data: TimeLogData,
  ) {}

  static create(data: TimeLogData): TimeLog {
    this.validate(data);
    const normalized = this.normalize(data);
    return new TimeLog(normalized);
  }

  private static validate(data: TimeLogData): void {
    const start = new Date(data.startTime);
    const end = new Date(data.endTime);

    if (isNaN(start.getTime()) || isNaN(end.getTime())) {
      throw new Error('Invalid date format in startTime or endTime.');
    }

    if (start > end) {
      throw new Error('startTime cannot be later than endTime.');
    }
  }

  private static normalize(data: TimeLogData): TimeLogData {
    const start = new Date(data.startTime);
    const end = new Date(data.endTime);

    return {
      ...data,
      startTime: start.toISOString(),
      endTime: end.toISOString(),
    };
  }
}
```